src/gui/mainwindow.cpp

```diff
@@ -1142,8 +1142,8 @@ void MainWindow::closeEvent(QCloseEvent *e)
 #else
     const bool goToSystrayOnExit = pref->closeToTray();
     if (!m_forceExit && m_systrayIcon && goToSystrayOnExit && !this->isHidden()) {
-        hide();
-        e->accept();
+        e->ignore();
```

Comment:
Can you verify it still works with `e->accept();` ?
I think it is weird to have `e->ignore();` here.

Reply:
It does not work with `e->accept();`. But I think ignoring the event is not that weird: https://doc.qt.io/qt-5/qcloseevent.html#details.

Reply:
Sorry, I don't understand an initial problem. Docs say:
>Close events contain a flag that indicates whether the receiver wants the widget to be closed or not. When a widget accepts the close event, it is hidden (and destroyed if it was created with the Qt::WA_DeleteOnClose flag).

Why we need call "hide()"? What is really happening if we just accept the event?

Reply:
When we just accept the event, the bug stays.

Reply:
>Probably the difference is how hide() is called (synchronous vs asynchronous).

Sorry, I mean why `hide()` is needed? According to the docs the widget should be hidden if we accept CloseEvent.

Reply:
When `accept` is called, it calls `hide` synchronously. Instead of this, we call `ignore` and we call `hide` asynchronously (with the help of a timer). I honestly don't know why the async version works but sync version doesn't.